sct.py

```diff
@@ -1352,5 +1357,37 @@ def clean_aws_mocks(test_id: str | None, all_mocks: bool, verbose: bool, dry_run
     AwsMock.clean(test_id=test_id, all_mocks=all_mocks, verbose=verbose, dry_run=dry_run)
 
 
+@cli.command("generate-pt-report", help="Generate parallel timelines representation for the SCT test events")
+@click.option("-t", "--test-id", help="Test ID to search in sct-results")
+@click.option("-d", "--logdir", type=click.Path(exists=True), help="Directory with sct-results folder")
+def generate_parallel_timelines_report(logdir: str | None, test_id: str | None) -> None:
+    add_file_logger()
+
+    event_log_file = "raw_events.log"
+
+    if not test_id:
+        test_id = os.environ.get("SCT_TEST_ID")
+    if not logdir:
+        logdir = os.environ.get('HOME')
+    LOGGER.debug("Searching for the required test run directory in %s...", logdir)
+    testrun_dir = get_testrun_dir(os.path.join(logdir, "sct-results"), test_id)
+    if not testrun_dir:
+        click.secho(message=f"Couldn't find directory for the required test run in '{logdir}'! Aborting...", fg="red")
+        sys.exit(1)
+    LOGGER.info("Found the test run directory '%s'", testrun_dir)
+
+    LOGGER.debug("Searching for the %s in %s...", event_log_file, testrun_dir)
+    events_log_path_list = list(Path(testrun_dir).glob(f"**/{event_log_file}"))
+    raw_events_log_path = events_log_path_list[0] if events_log_path_list else None
+
+    if raw_events_log_path:
```

Comment:
Since `Path.glob()` is a generator, it can be simplified to 

```python
if raw_events_log_path := next(Path(testrun_dir).glob(f"**/{event_log_file}"), None):
    ...
```

But actually it's better to handle bad situation first and remove one branch of `if ... else ...`:

```python
raw_events_log_path = next(Path(testrun_dir).glob(f"**/{event_log_file}"), None)

if raw_events_log path is None:
    ...
    sys.exit(1)

...
```

Reply:
Reworked that part. Thanks!